src/cryptoadvance/specter/server_endpoints/devices.py

```diff
@@ -81,6 +85,8 @@ def new_device_keys(device_type):
                 except:
                     err = _("Failed to parse these xpubs") + ":\n" + "\n".join(xpub)
                     break
+        if device_type == "electrum":
+            keys.append(Key.parse_xpub(request.form["xpub"]))
```

Comment:
`parse_xpubs`  catches exceptions and gives the user a nice error message. However when entering a wrong xpub, `parse_xpub` raises an exception that is not caught.

Reply:
I think if the `parse_xpub` method would raise `SpecterError` instead of Exceptions, we also would get better Error-messages?! In general, don't catch Exceptions if you don't know how such a potential issue arises. If you know, throw `SpecterError` with a message which even a user could understand.

Reply:
When entering a wrong xpub I get this:

<img width="601" alt="grafik" src="https://user-images.githubusercontent.com/70536101/164277987-c84ec530-b6d3-404e-a4e9-8910091a428f.png">

When entering a wrong xpub with manual configuration (which uses `parse_xpubs`), I get this:
<img width="686" alt="grafik" src="https://user-images.githubusercontent.com/70536101/164280098-3a4f0d58-27e8-4f0e-a227-06a5bfad2245.png">

I'd say both messages are not so great, but they don't really differ to me.